### PR TITLE
Coinbase Wallet: Add Support for Instant Onboarding mode

### PR DESCRIPTION
**Pull request by johanneskares**
This PR adds support for the instant onboarding mode of Coinbase Smart Wallet. It enables the same behaviour of the `Create Wallet` button on https://wallet.coinbase.com and skipping the first screen, where around 50% of my userbase clicked the wrong button. 

Use like this:

```typescript
(coinbaseWalletConnector as any).setInstantOnboarding(true);
await connectAsync({ connector: coinbaseWalletConnector });
```

https://github.com/user-attachments/assets/1d38ad7d-e78a-42c1-84ce-aad1117fba41

### CODE REVIEW

File: .changeset/big-radios-sip.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@wagmi/connectors": patch
+---
+
+Supports Instant Onboarding mode of Coinbase Smart Wallet
```
Comment (tmm) [inline .changeset/big-radios-sip.md:5]:
```suggestion
Added Coinbase Smart Wallet "Instant Onboarding" mode to `coinbaseWallet`.
```

File: .changeset/big-radios-sip.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@wagmi/connectors": patch
```
Comment (tmm) [inline .changeset/big-radios-sip.md:2]: New feature so `minor`

```suggestion
"@wagmi/connectors": minor
```

**Review by tmm — CHANGES_REQUESTED** [review]
With https://github.com/wevm/wagmi/pull/4453 merged, might be better to add an `instantOnboarding` property to `connector.connect` so it's discoverable and easier for folks to use.

```ts
const { connect, connectors } = useConnect()
connect({
  connector: connectors[0],
  instantOnboarding: true, // <-- `instantOnboarding` type inferred, shows up in autocomplete, etc.
})
```
```ts
const config = createConfig({
  chains: [mainnet],
  connectors: [coinbaseWallet({ ... })],
  transports: { [mainnet.id]: http() },
})

declare module 'wagmi' {
  interface Register {
    config: typeof config
  }
}
```

**Comment by johanneskares** [comment]
just realized @tmm you finished this. Thanks a log 🙏

**Pull request merged**